aws_lambda_powertools/utilities/parameters/ssm.py

```diff
@@ -0,0 +1,212 @@
+        >>> from aws_lambda_powertools.utilities.parameters import SSMProvider
+        >>>
+        >>> config = Config(region_name="us-west-1")
+        >>> ssm_provider = SSMProvider(config=config)
+        >>>
+        >>> ssm_provider.get("/my/parameter")
+
+    **Retrieves multiple parameter values from Systems Manager Parameter Store using a path prefix**
+
+        >>> from aws_lambda_powertools.utilities.parameters import SSMProvider
+        >>> ssm_provider = SSMProvider()
+        >>>
+        >>> ssm_provider.get_multiple("/my/path/prefix")
+
+    **Retrieves multiple parameter values from Systems Manager Parameter Store passing options to the SDK call**
+
+        >>> from aws_lambda_powertools.utilities.parameters import SSMProvider
+        >>> ssm_provider = SSMProvider()
+        >>>
+        >>> ssm_provider.get_multiple("/my/path/prefix", MaxResults=10)
+    """
+
+    client = None
+
+    def __init__(
+        self, config: Optional[Config] = None,
+    ):
+        """
+        Initialize the SSM Parameter Store client
+        """
+
+        config = config or Config()
+        self.client = boto3.client("ssm", config=config)
+
+        super().__init__()
+
+    def _get(self, name: str, decrypt: bool = False, **sdk_options) -> str:
+        """
+        Retrieve a parameter value from AWS Systems Manager Parameter Store
+
+        Parameters
+        ----------
+        name: str
+            Parameter name
+        decrypt: bool, optional
+            If the parameter value should be decrypted
+        sdk_options: dict, optional
+            Dictionary of options that will be passed to the get_parameter call
+        """
+
+        # Explicit arguments will take precedence over keyword arguments
+        sdk_options["Name"] = name
+        sdk_options["WithDecryption"] = decrypt
+
+        return self.client.get_parameter(**sdk_options)["Parameter"]["Value"]
+
+    def _get_multiple(self, path: str, decrypt: bool = False, recursive: bool = False, **sdk_options) -> Dict[str, str]:
+        """
+        Retrieve multiple parameter values from AWS Systems Manager Parameter Store
+
+        Parameters
+        ----------
+        path: str
+            Path to retrieve the parameters
+        decrypt: bool, optional
+            If the parameter values should be decrypted
+        recursive: bool, optional
+            If this should retrieve the parameter values recursively or not
+        sdk_options: dict, optional
+            Dictionary of options that will be passed to the get_parameters_by_path call
+        """
+
+        # Explicit arguments will take precedence over keyword arguments
+        sdk_options["Path"] = path
+        sdk_options["WithDecryption"] = decrypt
+        sdk_options["Recursive"] = recursive
+
+        response = self.client.get_parameters_by_path(**sdk_options)
+        parameters = response.get("Parameters", [])
+
+        # Keep retrieving parameters
+        while "NextToken" in response:
```

Comment:
I always recommend using actual paginators:

```python
response = self.client.get_paginator('get_parameters_by_path').paginate(**sdk_options)
```

so you don't have to reinvent the "while" :)